src/console/Program.cs

```diff
@@ -27,47 +30,95 @@ public class Program
         public static async Task Main()
         {
             var config = GetEnvironmentConfig();
+            var eventHubName = GetEventHubName(config);
 
-            // determine which event hub to read from
-            var eventHub = Environment.GetEnvironmentVariable("WEBJOBS_NAME");
-            if (eventHub == null)
-            {
-                eventHub = config.GetSection("Console:EventHub").Value;
-            }
-
-            System.Console.WriteLine($"Reading from event hub: {eventHub}");
-            System.Console.WriteLine($"Logs and Metrics will be written to Application Insights");
-            var eventHubOptions = GetEventHubInfo(config, eventHub);
+            var eventHubOptions = GetEventHubInfo(config, eventHubName);
 
             EnsureArg.IsNotNullOrWhiteSpace(eventHubOptions.EventHubConnectionString);
             EnsureArg.IsNotNullOrWhiteSpace(eventHubOptions.EventHubName);
 
-            var eventBatchingOptions = new EventBatchingOptions();
-            config.GetSection(EventBatchingOptions.Settings).Bind(eventBatchingOptions);
-
-            var serviceProvider = GetRequiredServiceProvider(config, eventHub);
+            var serviceProvider = GetRequiredServiceProvider(config, eventHubName);
             var logger = serviceProvider.GetRequiredService<ITelemetryLogger>();
-            var eventConsumers = GetEventConsumers(config, eventHub, serviceProvider, logger);
+
+            var eventConsumers = GetEventConsumers(config, eventHubName, serviceProvider, logger);
 
             var storageOptions = new StorageCheckpointOptions();
             config.GetSection(StorageCheckpointOptions.Settings).Bind(storageOptions);
-            storageOptions.BlobPrefix = eventHub;
-            var checkpointClient = new StorageCheckpointClient(storageOptions, logger);
 
+            var storageCheckpointClient = GetStorageCheckpointClient(storageOptions, logger, eventHubName);
             var eventConsumerService = new EventConsumerService(eventConsumers, logger);
 
+            var incomingEventReader = GetEventProcessorClient(storageCheckpointClient.GetBlobContainerClient(), eventHubOptions);
+            var eventHubReader = GetEventProcessor(config, eventConsumerService, storageCheckpointClient, logger);
+
+            System.Console.WriteLine($"Reading from event hub: {eventHubName}");
+            System.Console.WriteLine($"Logs and Metrics will be written to Application Insights");
             var ct = new CancellationToken();
+            await eventHubReader.RunAsync(incomingEventReader, ct);
+        }
+
+        public static string GetEventHubName(IConfiguration config)
+        {
+            var eventHub = Environment.GetEnvironmentVariable("WEBJOBS_NAME");
+            if (eventHub == null)
```

Comment:
eventHub is a string, we should probably use !string.IsNullOrWhitespace(EventHubs).